modules/services/window-managers/i3-sway/lib/options.nix

```diff
@@ -389,11 +388,7 @@ in {
       options = {
         titlebar = mkOption {
           type = types.bool;
-          default = !isGaps;
-          defaultText = if isI3 then
-            "xsession.windowManager.i3.package != nixpkgs.i3-gaps (titlebar should be disabled for i3-gaps)"
-          else
-            "false";
+          default = true;
```

Comment:
this seems to change the default for sway as is shown in the test. Is that because the default in sway/i3 changed ? otherwise remove "default" probably ?

Reply:
This was what I was wondering about previously.
After checking in the upstream projects code, the default value for those options is indeed `normal` (and not `pixel`) for both [`i3`](https://github.com/i3/i3/blob/9c8746c00fee18561e2a0a039d9b1c7387b848c5/src/config.c#L212-L213) and [`sway`](https://github.com/swaywm/sway/blob/88c17ece3b2617762c3f5f1c1fa79fe8f4df0082/sway/config.c#L294-L295).
The author of the sway module [confirmed that at the time he just wanted to copy what was done for `i3-gaps`](https://github.com/nix-community/home-manager/pull/3563#issuecomment-1373460575).

Setting `titlebar` to `true` is consistent with the default value used in `i3` and `sway`, i.e. `default_[floating_]border normal`.

Reply:
I completely ignored your previous posts very sorry. One bad example of context switching :( if normal is the default then yes we should have that.  https://github.com/nix-community/home-manager/pull/3563#issuecomment-1373460575 comments on using stateVersion and this seems like a good idea as it's easy to do. Pair it with a docs/release-notes/rl-2305.adoc entry (which can replace/complete the news.nix change)

Reply:
All right ! How should I handle the `stateVersion` check ?
If `stateVersion >= 23.05`, then put `*titlebar` settings default to `true` with `sway`, else `false` ?

Reply:
yes

Reply:
Which one seems the best to you ?

1. `default = !((versionOlder stateVersion "23.05") and isSway);`
2. `default = (versionAtLeast stateVersion "23.05") or isI3;`

Reply:
Neither preserve the original behaviour for people who had gaps activated right?

Reply:
You are right. If we go for preserving the previous behavior, let's do it seriously.
Latest push fixes that:

```
default = if versionOlder stateVersion "23.05"
  then (isI3 and (cfg.config.gaps == null))
  else true
;
```

Reply:
The tests are failing. What is the `stateVersion` used in the tests ?

Reply:
It seems that getting the stateVersion here:
```
  inherit (config.home) stateVersion;
```
Leads to the crash of `nix-shell . -A install`.
Other modules seem to do it succesfully...

It works if I hardcode it:
```
stateVersion = "20.09";
```

Reply:
Wow, this error was not easy to fix because the message was really misleading.
What caused the problem was adding a dependence on the `stateVersion` without providing a `defaultText`.
It was not obvious to me that the lack of `defaultText` lead to this error during `nix-shell . -A install`.

Anyway, I think everything is now correct.